### PR TITLE
CI: Debug min build timing out

### PR DESCRIPTION
**Pull request by mroeschke**
I've gathered that it's due to some S3 test and that now is the time to enforce running non-single tests with xdist and `-n auto`.

Will follow up with one single build after: https://github.com/pandas-dev/pandas/issues/44584

### CODE REVIEW

**Comment by mroeschke** [comment]
@jbrockmendel this PR will "solve" the min build issue if you think it's good to go

File: pandas/tests/io/parser/test_parse_dates.py [hunk 1]
```diff
@@ -948,7 +948,10 @@ def test_parse_dates_custom_euro_format(all_parsers, kwargs):
             )
 
 
-@xfail_pyarrow
+# TODO: Depending on what tests are selected (i.e. pytest marks are selected)
+#  this test can xpass (2020-02-12: Azure MacOS 3.10 & GHA Linux Datamanager 3.8 xpass)
```
Comment (jbrockmendel) [inline pandas/tests/io/parser/test_parse_dates.py:952]: 1) year should be 2022, not 2020?
2) are you sure this isn't just xpassing bc there's a new pyarrow version?

File: pandas/tests/io/json/test_compression.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def test_read_zipped_json(datapath):
 
 
 @td.skip_if_not_us_locale
+@pytest.mark.single_cpu
```
Comment (jbrockmendel) [inline pandas/tests/io/json/test_compression.py:41]: does this render the xfail unnecessary?

**Comment by jbrockmendel** [comment]
> Will follow up with one single build after: #44584

IIUC this means the marked tests won't be run at all on the CI until #44584?

**Comment by mroeschke** [comment]
> > Will follow up with one single build after: #44584
> 
> IIUC this means the marked tests won't be run at all on the CI until #44584?

Correct. If you feel more comfortable for me to add the single build in this PR I can.

File: pandas/tests/io/json/test_compression.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/io/json/test_compression.py:41]: Good catch. Will remove that

File: pandas/tests/io/parser/test_parse_dates.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/io/parser/test_parse_dates.py]: Good catch. Looks like it was a pyarrow version

**Comment by jbrockmendel** [comment]
> Correct. If you feel more comfortable for me to add the single build in this PR I can.

No, I'm fine with having it as a follow-up.

**Comment by mroeschke** [comment]
Merging to get to green, can follow up if needed

**Comment by mroeschke** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**